nemo/collections/common/tokenizers/sentencepiece_tokenizer.py

```diff
@@ -36,13 +37,19 @@ class SentencePieceTokenizer(TokenizerSpec, ChatTemplateMixin):
         special_tokens: either list of special tokens or dictionary of token name to token value
         legacy: when set to True, the previous behavior of the SentecePiece wrapper will be restored,
             including the possibility to add special tokens inside wrapper.
+        ignore_extra_whitespaces: whether to ignore extra whitespaces in the input text while encoding.
```

Comment:
I'm not sure I understand - why is this needed?

Reply:
This is for already trained tokenizers (which are trained to ignore extra whitespaces). by adding this parameter we can make inference to these models not to ignore them.
So, for example, if we want to encode ASR model's prediction `text with     a lot of spaces`, we will get this tokens (and their corresponding ids) `["_text", "_with", "_", "a", "_lot", "_of", "_spaces"]`.  but instead we should have got  `["_text", "_with", "_",  "_",  "_",  "_", "a", "_lot", "_of", "_spaces"]` which was in fact predicted by the model and can be used for alignment purposes (I believe there are other use-cases too)